src/vmemcache_heap.c

```diff
@@ -198,110 +230,186 @@ vmcache_heap_destroy(struct heap *heap)
 {
 	LOG(3, "heap %p", heap);
 
-	VEC_DELETE(&heap->entries);
 	util_mutex_destroy(&heap->lock);
 	Free(heap);
 }
 
+/*
+ * vmcache_free_small_extent -- (internal) free the smallest extent
+ */
+static int
+vmcache_free_small_extent(struct heap *heap, void *small_extent)
```

Comment:
`vmemcache_free_extent`.

Anyway, we have three different functions to remove an extent from a list with only slight variations. Is there a way to reduce the repetitions?